specs/eventing/control-plane.md

```diff
@@ -0,0 +1,985 @@
+rules:
+  - apiGroups: ["eventing.knative.dev"]
+    resources: ["broker", "trigger"]
+    verbs: ["get", "list", "create", "update", "patch", "delete"]
+  - apiGroups: ["messaging.knative.dev"]
+    resources: ["channel", "subscription"]
+    verbs: ["get", "list", "create", "update", "patch", "delete"]
+```
+
+In order to support resolving resources which meet the
+[Addressable](./overview.md#addressable) contract, the system controlling the
+[Trigger](#trigger-lifecycle) or [Subscription](#subscription-lifecycle) will
+need _read_ access to these resources. On Kubernetes, this is most easily
+achieved using role aggregation; on systems using Kubernetes RBAC, resources
+which wish to participate in Addressable resolution are expected to provide the
+following `ClusterRole`:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: <component>-addressable-resolver
+  labels:
+    duck.knative.dev/addressable: "true"
+rules:
+  - apiGroups: [<resource apiGroup>]
+    resources: [<resource kind >, <resource kind>/status]
+    verbs: ["get", "list", "watch"]
+```
+
+This configuration advice SHALL NOT indicate a requirement for Kubernetes RBAC
+support.
+
+<!-- TODO: define aggregated API roles
+
+Ref:
+- https://github.com/knative/specs/blob/main/specs/eventing/channel.md#aggregated-channelable-manipulator-clusterrole
+- https://github.com/knative/specs/blob/main/specs/eventing/channel.md#aggregated-addressable-resolver-clusterrole
+-
+-->
+
+## Error Signalling
+
+See [the Knative common condition guidance](../common/error-signalling) for how
+resource errors are signalled to the user.
+
+## Resource Lifecycle
+
+### Broker Lifecycle
+
+A Broker represents an Addressable endpoint (i.e. it MUST have a
+`status.address.url` field) which can receive, store, and forward events to
+multiple recipients based on a set of attribute filters (Triggers). Triggers are
+associated with a Broker based on the `spec.broker` field on the Trigger; it is
+expected that the controller for a Broker will also control the associated
+Triggers. When the Broker's `Ready` condition is `true`, the Broker MUST provide
+a `status.address.url` which accepts all valid CloudEvents and MUST attempt to
+forward the received events for filtering to each associated Trigger whose
+`Ready` condition is `true`. As described in the
+[Trigger Lifecycle](#trigger-lifecycle) section, a Broker MAY forward events to
+an associated Trigger which which does not currently have a `true` `Ready`
+condition, including events received by the Broker before the Trigger was
+created.
+
+The annotation `eventing.knative.dev/broker.class` MAY be used to select a
+particular implementation of a Broker. When a Broker is created, the
+`eventing.knative.dev/broker.class` annotation and the `spec.config` field MUST
+be populated (`spec.config` MAY be an empty object) to indicate which of several
+possible Broker implementations to use. It is RECOMMENDED to default the
```

Comment:
This RECOMMENDED feels like it's inconsistent with the previous MUST.

Side question... why MUST these fields be populated? Why can't the implementation simply have it's own default value that it knows about behind the scenes? Why must it be exposed to the user?

Reply:
I think the RECOMMENDED is still inconsistent with the MUST.
What happens if the values are invalid? Do we block the creation in etcd or expose an error status.condition?

Reply:
(relaxed to "SHOULD be populated")